format/tmux/formater_test.go

```diff
@@ -156,3 +161,148 @@ func TestFormater_divergence(t *testing.T) {
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`#\[fg=default]⎇ #\[fg=default][\w\/.-]+..(#\[fg=default][\w\/.-]+)?#\[fg=default] - #\[fg=default].+`),
+		},
+		{
+			name: "default format with diff delimiters",
+			styles: styles{
+				Clean: "CleanStyle",
+			},
+			symbols: symbols{
+				Branch: "⎇ ",
+				Clean:  "CleanSymbol",
+			},
+			display: []string{"branch", "~~", "remote", " | ", "flags"},
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`#\[fg=default]⎇ #\[fg=default][\w\/.-]+~~(#\[fg=default][\w\/.-]+)?#\[fg=default] | #\[fg=default].+`),
+		},
+		{
+			name: "no branch or delimiter0",
+			styles: styles{
+				Clean: "CleanStyle",
+			},
+			symbols: symbols{
+				Branch: "⎇ ",
+				Clean:  "CleanSymbol",
+			},
+			display: []string{"remote", " - ", "flags"},
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`(#\[fg=default][\w\/.-]+)?#\[fg=default] - #\[fg=default].+`),
+		},
+		{
+			name: "no remote and delimiter0",
+			styles: styles{
+				Clean: "CleanStyle",
+			},
+			symbols: symbols{
+				Branch: "⎇ ",
+				Clean:  "CleanSymbol",
+			},
+			display: []string{"branch", " - ", "flags"},
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`#\[fg=default]⎇ #\[fg=default][\w\/.-]+ - #\[fg=default].+`),
+		},
+		{
+			name: "branch only",
+			styles: styles{
+				Clean: "CleanStyle",
+			},
+			symbols: symbols{
+				Branch: "⎇ ",
+				Clean:  "CleanSymbol",
+			},
+			display: []string{"branch"},
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`#\[fg=default]⎇ #\[fg=default][\w\/.-]+`),
+		},
+		{
+			name: "remote only",
+			styles: styles{
+				Clean: "CleanStyle",
+			},
+			symbols: symbols{
+				Branch: "⎇ ",
+				Clean:  "CleanSymbol",
+			},
+			display: []string{"remote"},
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`(#\[fg=default][\w\/.-]+)?#\[fg=default]`),
+		},
+		{
+			name: "flags only",
+			styles: styles{
+				Clean: "CleanStyle",
+			},
+			symbols: symbols{
+				Branch: "⎇ ",
+				Clean:  "CleanSymbol",
+			},
+			display: []string{"flags"},
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`#\[fg=default].+`),
+		},
+		{
+			name: "no delimiters",
+			styles: styles{
+				Clean: "CleanStyle",
+			},
+			symbols: symbols{
+				Branch: "⎇ ",
+				Clean:  "CleanSymbol",
+			},
+			display: []string{"branch", "remote", "flags"},
+			st: &gitstatus.Status{
+				IsClean: true,
+			},
+			want: regexp.MustCompile(`#\[fg=default]⎇ #\[fg=default][\w\/.-]+(#\[fg=default][\w\/.-]+)?#\[fg=default]#\[fg=default].+`),
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			f := &Formater{
+				Config: Config{Styles: tc.styles, Symbols: tc.symbols, Display: tc.display},
+				st:     tc.st,
+			}
+			st, _ := gitstatus.New()
```

Comment:
Can we do like in the other tests (i.e without having to run `git` under the hood)?

Reply:
I mean. Tests should not depend on the git status of current directory, tests should only be test formatting of a specially crafted git.Status struct with the tmux formatter